src/diagnostics.rs

```diff
@@ -671,6 +685,79 @@ impl DiagnosticReporter for CodeSpanDiagnosticReporter {
     }
 }
 
+/// a DiagnosticReporter that reports diagnostics using clang-format
+pub struct ClangFormatDiagnosticReporter {
+    files: SimpleFiles<String, String>,
+}
+
+impl ClangFormatDiagnosticReporter {
+    fn new() -> Self {
+        ClangFormatDiagnosticReporter {
+            files: SimpleFiles::new(),
+        }
+    }
+}
+
+impl Default for ClangFormatDiagnosticReporter {
+    fn default() -> Self {
+        ClangFormatDiagnosticReporter::new()
+    }
+}
+
+impl DiagnosticReporter for ClangFormatDiagnosticReporter {
+    fn report(&self, diagnostics: &[AssessedDiagnostic], file_id: usize) {
+        for ad in diagnostics {
+            let d = &ad.diagnostic;
+            let location = d.get_location();
+            let file_name = match self.files.get(file_id) {
+                Ok(v) => v.name(),
+                Err(err) => {
+                    eprintln!("Unable to find affected file : {}", err);
+                    "unknown"
+                }
+            };
+
+            let start_location = get_location(&self.files, file_id, location.get_start());
+            let end_location = get_location(&self.files, file_id, location.get_end());
+
+            // print diagnostic
+            // file-name:{range}: severity: msg
+            eprintln!(
+                "{}:{{{}:{}-{}:{}}}: {}: {}",
```

Comment:
We could probably use some kind of steam builder here to add items only conditionally (If location is not avaiable we skip it, if no file is given we add a generic error)
Also we need to test different scenarios here.